ops/model.py

```diff
@@ -1011,16 +1011,18 @@ def location(self):
 class Container:
     """Represents a named container in a unit.
 
+    This class should not be instantiated directly, instead use :meth:`Unit.get_container`
+    or :attr:`Unit.containers`.
+
     Attributes:
         name: The name of the container from metadata.yaml (eg, 'postgres').
     """
 
-    def __init__(self, name, pebble_client=None):
+    def __init__(self, name, backend, pebble_client=None):
         self.name = name
 
         if pebble_client is None:
-            socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
-            pebble_client = pebble.Client(socket_path=socket_path)
+            pebble_client = backend.get_pebble(name)
```

Comment:
The name doesn't seem enough to determine which pebble is to be used on the local system. It is assuming that it is a container, but pebble doesn't require containers to actually work.  It need to either take a socket, or something else that enables us to distinguish on the backend side what to do with that name.

Reply:
Good call. Fix in https://github.com/canonical/operator/pull/473 -- just moving the container-specific building of `socket_path` from `_ModelBackend.get_pebble` to `Container.__init__`.

Reply:
Thanks!